### ISSUE TITLE
Rule proposal: `no-use-v-if-with-v-for`

### ISSUE THREAD

**Issue by ota-meshi** [vuejs/eslint-plugin-vue#403]
**Please describe what the rule should do:**

This rule warns never use v-if on the same element as v-for.

Style guide:
https://vuejs.org/v2/style-guide/#Avoid-v-if-with-v-for-essential

**What category of rule is this? (place an "X" next to just one item)**

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```xml
<ul>
  <li
    v-for="user in users"
    v-if="user.isActive"
    :key="user.id"
  >
    {{ user.name }}
  </li>
</ul>
```

**Why should this rule be included?**

This rule is stated in the Style guide as ESSENTIAL.
It is similar to [`vue/no-confusing-v-for-v-if`](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/no-confusing-v-for-v-if.md), but the content of `code examples` is not warned.

I confirmed issues, but I could not find duplicate rules.
I'm sorry if it was duplicated.

**Comment by mysticatea** [comment]
Thank you for this issue.

Looks good to me.
Maybe the existing `vue/no-confusing-v-for-v-if` rule should be deprecated after we add `vue/no-use-v-if-with-v-for` rule.

cc: @chrisvfritz @michalsnik

**Comment by ota-meshi** [comment]
Thank you for considering the proposal !!

**Comment by chrisvfritz** [comment]
I like it too. 🙂

**Comment by kenneyHong** [comment]
> I like it too. 🙂
I do not like(我不喜欢)
